samples/net/dsa/src/dsa_lldp.c

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright (c) 2020 DENX Software Engineering GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+
+#include <net/socket.h>
+#include <net/net_core.h>
+#include <net/net_l2.h>
+#include <net/net_if.h>
+#include <net/socket.h>
+#include <net/ethernet.h>
+#include <net/lldp.h>
+#include <errno.h>
+
+#include <logging/log.h>
+#include <malloc.h>
+#include "main.h"
+
+/* Loglevel of dsa_lldp function */
+LOG_MODULE_REGISTER(dsa_lldp, LOG_LEVEL_DBG);
+
+#define LLDP_ETHER_TYPE 0x88CC
+#define DSA_BUF_SIZ 128
+int dsa_lldp_send(struct net_if *iface, struct instance_data *pd,
+		  uint16_t lan, int src_port, int origin_port, int cmd,
+		  struct eth_addr *origin_addr)
+{
+	int ret, len;
+	char buffer[DSA_BUF_SIZ], s[24];
+	struct sockaddr_ll dst;
+	struct ethernet_context *ctx = net_if_l2_data(iface);
+	struct net_eth_hdr *eth_hdr = (struct net_eth_hdr *) buffer;
+	uint8_t *p = &buffer[sizeof(struct net_eth_hdr)];
+	uint8_t *pb = p;
+
+	lan = ctx->dsa_port_idx;
+
+	dst.sll_ifindex = net_if_get_by_iface(iface);
+	/* Construct the Ethernet header */
+	memset(buffer, 0, DSA_BUF_SIZ);
+	/* Ethernet header */
+	/* Take MAC address assigned to LAN port */
+	memcpy(eth_hdr->src.addr, net_if_get_link_addr(iface)->addr, ETH_ALEN);
+	eth_hdr->dst.addr[0] = MCAST_DEST_MAC0;
+	eth_hdr->dst.addr[1] = MCAST_DEST_MAC1;
+	eth_hdr->dst.addr[2] = MCAST_DEST_MAC2;
+	eth_hdr->dst.addr[3] = MCAST_DEST_MAC3;
+	eth_hdr->dst.addr[4] = MCAST_DEST_MAC4;
+	eth_hdr->dst.addr[5] = MCAST_DEST_MAC5;
+
+	/* Ethertype field */
+	eth_hdr->type = htons(LLDP_ETHER_TYPE);
+
+	/* LLDP packet data */
+	/* Chasis ID */
+	dsa_buf_write_be16((LLDP_TLV_CHASSIS_ID << 9) | (ETH_ALEN + 1), &p);
+	*p++ = 4; /* subtype */
+	memcpy(p, net_if_get_link_addr(iface)->addr, ETH_ALEN);
+	p += ETH_ALEN;
+
+	/* PORT ID */
+	dsa_buf_write_be16((LLDP_TLV_PORT_ID << 9) | (ETH_ALEN + 1), &p);
+	*p++ = 3; /* subtype */
+	memcpy(p, net_if_get_link_addr(iface)->addr, ETH_ALEN);
+	p += ETH_ALEN;
+
+	/* TTL ID */
+	dsa_buf_write_be16((LLDP_TLV_TTL << 9) | 2, &p);
+	*p++ = 0; /* TTL field is 2 bytes long */
+	*p++ = 120;
+
+	/* SYSTEM NAME */
+	memset(s, 0, sizeof(s));
+	sprintf(s, "ip_k66f LAN:%d", lan);
+	dsa_buf_write_be16((LLDP_TLV_SYSTEM_NAME << 9) | strlen(s), &p);
+	memcpy(p, s, strlen(s));
+	p += strlen(s);
+
+	len = sizeof(struct net_eth_hdr) + (p - pb);
+	ret = sendto(pd->sock, buffer, len, 0, (const struct sockaddr *)&dst,
+		     sizeof(struct sockaddr_ll));
+	if (ret < 0) {
+		LOG_ERR("Failed to send, errno %d", errno);
+	}
+
+	return 0;
+}
+
+void dsa_lldp_print_info(uint8_t *lldp_p, uint8_t lanid)
+{
+	uint16_t tl, length;
+	uint8_t type, subtype;
+	uint8_t *p, t1, t2;
+	char *t;
+
+	LOG_INF("LLDP pkt recv -> lan%d", lanid);
```

Comment:
Use `LOG_DBG` as this is clearly a debug msg (and we have dbg level set anyway)
Same things to INF calls below.

Reply:
If you don't mind - I would opt for keeping this,as it is part of the example/test. 

With `LOG_INF` I can see the MAC address of received packet and print information from which KSZ8794 LAN port the LLDP packet has been read.
This allows me to check (with e.g. wireshark/tcpdump) if the "port separation" feature (as configured in this example) works as expected.
Moreover, with this information I can check if board did not hang.

Reply:
> If you don't mind - I would opt for keeping this,as it is part of the example/test.

No worries, we can keep the INF level.